test/hotspot/jtreg/compiler/vectorapi/TestVectorSlice.java

```diff
@@ -0,0 +1,918 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.vectorapi;
+
+import compiler.lib.ir_framework.*;
+import jdk.incubator.vector.*;
+import jdk.test.lib.Asserts;
+
+/**
+* @test
+* @bug 8287835
+* @summary Test vector slice intrinsics
+* @modules jdk.incubator.vector
+* @requires vm.compiler2.enabled
+* @library /test/lib /
+* @run driver compiler.vectorapi.TestVectorSlice
+*/
+public class TestVectorSlice {
+    @Run(test = {"testB64", "testB128", "testB256", "testB512"})
+    static void testBytes() {
+        int maxSize = 64;
+        int diff = 70;
+        byte[][] dst = new byte[maxSize + 1][maxSize];
+        byte[] src1 = new byte[maxSize];
+        byte[] src2 = new byte[maxSize];
+        for (int i = 0; i < src1.length; i++) {
+            src1[i] = (byte)i;
+            src2[i] = (byte)(i + diff);
+        }
+
+        int length = 8;
+        testB64(dst, src1, src2);
+        for (int i = 0; i <= length; i++) {
+            for (int j = 0; j < length; j++) {
+                int offset = i + j;
+                byte expected;
+                if (offset < length) {
+                    expected = (byte)offset;
+                } else {
+                    expected = (byte)(offset + diff - length);
+                }
+                Asserts.assertEquals(expected, dst[i][j]);
+            }
+        }
```

Comment:
It should be possible to factor out this code into something like this:
```
  assertOffsets(length, (expected, i, j) -> Assert.assertEquals((byte)expected, dst[i][j])
```

Reply:
Fixed.